src/components/library/CitationSearch.tsx

```diff
@@ -206,9 +224,13 @@ export const CitationSearch: React.FC<{
     return handleCite(items)
   }, [handleCite, selected])
 
-  const handleQuery = useCallback((event: ChangeEvent<HTMLInputElement>) => {
-    setQuery(event.target.value)
-  }, [])
+  const debouncedCallback = debounce((value) => setQuery(value.trim()), 800)
+
+  const handleQuery = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) =>
+      debouncedCallback(event.target.value),
```

Comment:
That's a rather generic and not very meaningful naming

Reply:
That can be fixed while in qa. Resolving

Reply:
do you have naming in your mind, so we could use it

Reply:
how about debouncedSetQuery?

Reply:
renamed